5_models/Modeling_Lesson.Rmd

```diff
@@ -0,0 +1,224 @@
+
+```
+
+Example of an anova
+```{r}
+aov(dprime ~ adhdF, data = df)
+
+```
+
+### To add multiple factors 
+(in anova or linear regression), simply add  a "+" for each variable on the right of the formula 
+
+*Y ~ X1 + X2 + X3*
+
+Here is a linear model example with 2 predictors (continuous)
+```{r}
+lm(dprime ~ adhd + bis, data = df)
+
+```
+
+Here is an ANOVA example with 2 predictors (continuous)
+```{r}
+aov(dprime ~ adhdF + bis, data = df)
+
+```
+
+### Interactions 
+When you want to test the interaction between two variables, add a ":". 
+The colon will automatically create an interaction regressor between x1 and x2.
+You still need to add each variable alone to calculate their main effects. 
+
+Y ~ X1 + X2 + X1:X2
+
+Alternatively, a shortcut is to use an asterix "*" that provides a shortcut. It will automatically create an interaction regressor AND the main effects for each variable.
+
+ Y ~ X1*X2  is identical to Y ~ X1 + X2 + X1:X2
+
+Example of an interaction between 2 continous variables in linear regression:
+```{r}
+lm(dprime ~ adhd*bis, data = df)
+
+```
+
+Example of an interaction between 2 categorical variables in anova
+```{r}
+aov(dprime ~ adhdF*groupStatus, data = df)
+
+```
+
+In the case where there is no IV/DV (i.e. correlation between 2 continuous variables), there is no "~".
+Instead there is a ","" to delineate 2 different variables.
+
+Example of a correlation
+```{r}
+cor.test(df$dprime, df$adhd)
+
+```
+
+
+## Output of models 
+The output of your code will depend on the model type. Linear models have much more information to output relative to t-tests, so they require an additional "summary()" call on the model to see the full results. 
+
+### Linear models
+the output includes the following:
+a) the formula of your model 
+b) residuals
+c) coefficients of the model (estimate, st. error, t-value, p-value)
+d) overall model results (RSE, R-squared, F-statistics, p-value)
+
+```{r}
+# first, we will save our model object and call it lm1
+lm1 <-  lm(dprime ~ adhd, data = df)
+# then we will ask for the summary of the model results
+summary(lm1)
+```
+
+We can index specific components of the output. What if I want to save the coefficients to a table?
+```{r}
+summary(lm1)$coefficients
+
+# save the table for later
+my_table <- summary(lm1)$coefficients
+save(my_table, file = "my_table.Rda")
+```
+
+### Anova
+The output includes the results for each factor in the model.
+
+Here is an example for a one-way anova
+```{r}
+
+a1 <- aov(dprime ~ adhdF, data = df)
+
+summary(a1)
+```
+
+Here is an example for a 2x2 anova
+```{r}
+
+a2 <- aov(dprime ~ adhdF * groupStatus, data = df)
+
+summary(a2)
+```
+
+## Visualizing models 
+
+There are many ways (and packages) to plot the output of your models. 
+We will start with a handy package that plots the effects of many model types.
+
+Effect Displays for Linear, Generalized Linear, and Other Models
+#https://cran.r-project.org/web/packages/effects/effects.pdf
```

Comment:
we can change this to a markdown-formatted hyperlink? I think this used to be a comment inside a code chunk, but it looks like it's outside a code chunk now so the single # will cause this to render as a level 1 header as is.